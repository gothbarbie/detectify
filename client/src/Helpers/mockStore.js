import configureStore from 'redux-mock-store' //ES6 modules
import thunk from 'redux-thunk'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

export default mockStore
