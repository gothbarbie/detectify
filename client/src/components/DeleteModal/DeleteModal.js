import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { deleteNoteAndHideModalThunk } from '../../actions/notesActions'
import { hideModal } from '../../actions/modalActions'

import Button from '../Button'

const Title = styled.h2`
  font-size: 1.6rem;
  color: #666;
  font-weight: 600;
  margin: 0;
  text-align: center;
`

const Text = styled.div`
  font-size: 1.4rem;
  color: #666;
  font-weight: 400;
  margin: 3rem 0;

  span {
    color: #ba3333;
  }
`

const ButtonsWrapper = styled.div`
  display: grid;
  width: 100%;
  grid-template-columns: 1fr 1fr;
  grid-gap: 3rem;
`

const DeleteModal = ({
  deleteNoteAndHideModalThunk,
  hideModal,
  note,
  deleteNoteId,
}) => (
  <div>
    <Title>{note.title}</Title>
    <Text>
      Are you sure you want to <span>delete</span> this note?
    </Text>
    <ButtonsWrapper>
      <Button onClick={hideModal}>Cancel</Button>
      <Button
        onClick={() => deleteNoteAndHideModalThunk({ id: deleteNoteId })}
        type="red"
      >
        Delete
      </Button>
    </ButtonsWrapper>
  </div>
)

DeleteModal.propTypes = {
  deleteNoteAndHideModalThunk: PropTypes.func.isRequired,
  deleteNoteId: PropTypes.string.isRequired,
  hideModal: PropTypes.func.isRequired,
  note: PropTypes.shape({
    title: PropTypes.string.isRequired,
    timestamp: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
  }).isRequired,
}

const mapStateToProps = state => {
  const deleteNoteId = state.modal.deleteNoteId
  const note = state.notes[deleteNoteId]
  return { note, deleteNoteId }
}

const mapDispatchToProps = {
  hideModal,
  deleteNoteAndHideModalThunk,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(DeleteModal)
