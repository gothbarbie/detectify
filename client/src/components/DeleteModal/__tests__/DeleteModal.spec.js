import React from 'react'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'

import mockStore from '../../../Helpers/mockStore'

import DeleteModal from '../DeleteModal'

describe('Components/DeleteModal', () => {
  it('renders', () => {
    const modal = {
      show: true,
      deleteNoteId: 'n01',
    }
    const notes = {
      n01: {
        title: 'Note Title',
        timestamp: '',
        description: '',
        category: '',
        content: '',
      },
    }

    const initialState = {
      modal,
      notes,
    }

    const store = mockStore(initialState)

    const props = {
      deleteNoteAndHideModalThunk: jest.fn(),
      hideModal: jest.fn(),
    }
    const wrapper = shallow(
      <Provider store={store}>
        <DeleteModal {...props} store={store} />
      </Provider>
    ).dive()

    expect(wrapper).toMatchSnapshot()
  })
})
