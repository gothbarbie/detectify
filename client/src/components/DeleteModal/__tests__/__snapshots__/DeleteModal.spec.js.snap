// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/DeleteModal renders 1`] = `
<DeleteModal
  deleteNoteAndHideModalThunk={[Function]}
  hideModal={[Function]}
  modal={
    Object {
      "deleteNoteId": "n01",
      "show": true,
    }
  }
  notes={
    Object {
      "n01": Object {
        "category": "",
        "content": "",
        "description": "",
        "timestamp": "",
        "title": "Note Title",
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
